Add large width on `/sites` and `/domains/manage`

roo2 opened the pull request:
Adds max width to the sites and domains table 

Fixes 7186-gh-Automattic/dotcom-forge

1441px:

<img width="1199" alt="Screenshot 2024-05-16 at 11 49 04 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/6aedb365-0421-4e4a-bf28-f8600c4de04c">

1800px:

<img width="1193" alt="Screenshot 2024-05-16 at 11 49 15 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/8b87a239-4831-4387-9d77-84d0bbaf72cf">

I've also applied a max width to the domains page to max.

I had to jump through a few hoops to implement this, in part this was to maintain the header effect where the line runs right across the page and keep everything lined up properly.

### Testing instructions

Test different widths and page states on the /sites page.

roo2 committed:
add large width on /sites and /domains/manage

roo2 committed:
fixes

fushar reviewed (commented):
I think we need some left and right padding; it looks awkward on wider screens. The paddings are there in production.

<img width="1380" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/6a11fee6-ca4d-43d0-82a4-5db04177f2e8">

cleacos commented:
I left a comment in the issue

roo2 committed:
only apply at px

roo2 committed:
update domains

roo2 commented:
I've updated it so that the max width is only applied after >1920px wide screens, the table will be limited to 1400px and also added back in the side padding at this width, and tweaked alignment of the page number selector and header
<img width="1100" alt="Screenshot 2024-05-17 at 11 59 56 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/f0ca901c-4ed5-4926-a722-d1b5f4374bf1">

fushar reviewed (approved):
It looks good to me now 👍

cleacos reviewed (changes requested):
The table header is not aligned:

**Horizon:**
![image](https://github.com/Automattic/wp-calypso/assets/9832440/e5d66544-25a5-426e-8125-331d6a7d12f8)

**This PR:**
![image](https://github.com/Automattic/wp-calypso/assets/9832440/6a72ee11-976a-409d-9193-3a126a429e7d)

roo2 commented:
ahh thanks @cleacos , can't believe I missed that in my own screenshot! XD

roo2 committed:
fix misalignment

roo2 commented:
Ahh I see I accidentally removed a first-child selector for the `td` padding, I've fixed it now:

<img width="973" alt="Screenshot 2024-05-20 at 9 26 39 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/a7aa88e1-1178-4633-a2eb-766dff3ccdc5">
